src/OrleansRuntime/MultiClusterNetwork/MultiClusterOracle.cs

```diff
@@ -444,270 +443,261 @@ private void PublishMyStatusToNewDestinations(MultiClusterData delta)
+
         // numbering for tasks (helps when analyzing logs)
         private int idCounter;
 
-        private void PublishGossipToSilo(SiloAddress destinationSilo, MultiClusterData delta)
+        private SiloGossipWorker GetSiloWorker(SiloAddress silo)
         {
-            GossipStatus status;
-            if (!this.siloTasks.TryGetValue(destinationSilo, out status))
-                this.siloTasks[destinationSilo] = status = new GossipStatus();
+            SiloGossipWorker worker;
+            if (!this.siloWorkers.TryGetValue(silo, out worker))
+                this.siloWorkers[silo] = worker = new SiloGossipWorker(this, silo);
+            return worker;
+        }
+        private SiloGossipWorker GetClusterWorker(string cluster)
+        {
+            SiloGossipWorker worker;
+            if (!this.clusterWorkers.TryGetValue(cluster, out worker))
+                this.clusterWorkers[cluster] = worker = new SiloGossipWorker(this, cluster);
+            return worker;
+        }
+        private ChannelGossipWorker GetChannelWorker(IGossipChannel channel)
+        {
+            ChannelGossipWorker worker;
+            if (!this.channelWorkers.TryGetValue(channel, out worker))
+                this.channelWorkers[channel] = worker = new ChannelGossipWorker(this, channel);
+            return worker;
+        }
 
-            int id = ++this.idCounter;
-            logger.Verbose("-{0} PublishGossipToSilo {1} {2}", id, destinationSilo, delta);
+        // superclass for gossip workers.
+        // a gossip worker queues (push) and (synchronize) requests, 
+        // and services them using a single async worker
+        private abstract class GossipWorker : BatchWorker
+        {
+            public GossipWorker(MultiClusterOracle oracle)
+            {
+                this.oracle = oracle;
+            }
+            protected MultiClusterOracle oracle;
 
-            // the task that actually publishes
-            Func<Task, Task> publishAsync = async (Task prev) =>
+            // add all data to be published into this variable
+            protected MultiClusterData toPublish = new MultiClusterData();
+
+            // set this flag to request a full gossip (synchronize)
+            protected bool doSynchronize = false;
+
+            public void Publish(MultiClusterData data)
             {
-                await prev; // wait for previous publish to same silo
-                status.LastUse = DateTime.UtcNow;
-                status.Pending = true;
-                status.Address = destinationSilo;
-                try
-                {
-                    // publish to the remote system target
-                    var remoteOracle = InsideRuntimeClient.Current.InternalGrainFactory.GetSystemTarget<IMultiClusterGossipService>(Constants.MultiClusterOracleId, destinationSilo);
-                    await remoteOracle.Publish(delta, false);
+                toPublish = toPublish.Merge(data);
```

Comment:
Do I understand it correctly that this is the key line - for merging multiple publish requests without creating a new task, so that multiple updates can be sent together as part of the next task? But we don't impose any limit of a max number of merges here? Which is probably fine.

Maybe add logging around, so that the functionality is exposed at run time.